src/Cardano/Wallet/Kernel/DB/HdWallet/Derivation.hs

```
@@ -29,4 +34,21 @@ deriveIndex pickRange mkA hardeningMode =
                      HardDerivation -> (firstHardened, maxBound)
     in mkA <$> pickRange range
 
+-- Which derivation scheme is being used
+data DerivationScheme
+    -- Derivation scheme random is following bip32 scheme, ed25519v0 curve and addresses contain: account index and address index
+    = RandomDerivationScheme
+    -- Derivation scheme sequential is bip44 scheme, ed25519v1 curve and addresses contain no address payload. Root key derivation from mnemonic keys also differs from root key derivation in random scheme
+    | SequentialDerivationScheme
+    deriving (Show, Eq, Ord)
 
+derivationScheme :: Core.Address -> DerivationScheme
```

this can be removed now as this information is contained in wallet type and in db - will remove

---

yes - this could be used in tests at least. Additionally we could throw an exception if (or an error) if that invariant is not satisfied as a "failsafe" if something whent wrong.